surface/src/simplification_remove_unused_vertices.cpp

```diff
@@ -91,12 +91,12 @@ pcl::surface::SimplificationRemoveUnusedVertices::simplify(const pcl::PolygonMes
 
   // copy mesh information (and update indices)
   output.polygons.reserve (input.polygons.size ());
-  for (size_t polygon = 0; polygon < input.polygons.size (); ++polygon)
+  for (const auto &polygon : input.polygons)
   {
     pcl::Vertices corrected_polygon;
-    corrected_polygon.vertices.resize (input.polygons[polygon].vertices.size ());
-    for (size_t point = 0; point < input.polygons[polygon].vertices.size(); ++point)
-      corrected_polygon.vertices[point] = new_indices[input.polygons[polygon].vertices[point]];
+    corrected_polygon.vertices.resize (polygon.vertices.size ());
+    for (size_t point = 0; point < polygon.vertices.size(); ++point)
+      corrected_polygon.vertices[point] = new_indices[polygon.vertices[point]];
```

Comment:
This block would benefit from replacing the `corrected_polygon.vertices.resize()` for a call to `reserve()`, followed by a for-range loop of push_backs into `corrected_polygon.vertices`.

Reply:
It's increases readability, yes. But I'm not sure about the performance, because `push_back` needs to check always if size still fits and increase counter of used items, during the old code just accessed the item.

Thats why I don't apply:
```
    corrected_polygon.vertices.reserve (polygon.vertices.size ());
    for (const auto &vertex : polygon.vertices)
      corrected_polygon.vertices.emplace_back(new_indices[vertex]);
```

Reply:
My comment was not aimed at readability but at avoiding the value-initialization stage that takes place during `resize`.

Reply:
`vertices` just contains `int`, so it's not expensive to create them with using their value.

Reply:
Valid argument. The trade-off is most likely negligible, so let's go with your approach.